src/Toolkit/src/Command/UxToolkitInstallCommand.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Toolkit\Command;
+
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\UX\Toolkit\Compiler\Exception\TwigComponentAlreadyExist;
+use Symfony\UX\Toolkit\Compiler\TwigComponentCompiler;
+use Symfony\UX\Toolkit\ComponentRepository\CurrentTheme;
+use Symfony\UX\Toolkit\Registry\RegistryFactory;
+
+/**
+ * @author Jean-François Lépine
+ *
+ * @internal
+ */
+#[AsCommand(
+    name: 'ux:toolkit:install',
+    description: 'This command will install a new UX Component in your project',
+)]
+class UxToolkitInstallCommand extends Command
+{
+    public function __construct(
+        private readonly CurrentTheme $currentTheme,
+        private readonly RegistryFactory $registryFactory,
+        private readonly TwigComponentCompiler $compiler,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->addArgument('component', InputArgument::REQUIRED, 'The component name (Ex: button) or repository URL')
+            ->addOption(
+                'destination',
+                'd',
+                InputArgument::OPTIONAL,
+                'The destination directory',
+                'templates/components'
+            )
+            ->addOption('overwrite', 'o', InputOption::VALUE_NONE, 'Overwrite the component if it already exists')
```

Comment:
More generally, I think users are more used to `--force` option (also, this is what we use in `LockIconsCommand`:
```suggestion
            ->addOption('force', 'f', InputOption::VALUE_NONE, 'Force re-installation if already installed')
```